### PR TITLE
feat: [SRM-15101]: Azure Log Analytics sample data request changes

### PR DESCRIPTION
**Pull request by karansaraswat19**
## Describe your changes
- Added AzureLogsSampleDataRequest class to get sample log records
- Followed the steps to add a new health source extending common health source

## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50480)

### CODE REVIEW

**Comment by karansaraswat19** [comment]
trigger uts

**Review by deepak-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50480)* status: 17 file(s) to review. 

___
*[`srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/cvng/beans/azure/AzureLogsSampleDataRequest.java` line 67 at r1](https://reviewable.io/reviews/harness/harness-core/50480#-N_HgWYN2erM9q-KtcTH:-N_HgWYOCs1EY5d9cNCN:b2topa6) ([raw file](https://github.com/harness/harness-core/blob/24eae0635576cabc3b1971acc6cb5a8b43ed014e/srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/cvng/beans/azure/AzureLogsSampleDataRequest.java#L67)):*
> ```Java
>       throw new InvalidRequestException("Please configure the connector with client secret");
>     }
>     Map<String, Object> dslEnvVariables = super.fetchDslEnvVariables();
> ```

nit: Either move this to AbstractAzureDataRequest or AzureUtils. Same code will be used for different request.

**Comment by n00bitax** [comment]
Approved conditionally, fix it in next PR

- Handle gracefully when Auth mode in no token.
- Use RELEASE version instead of SNAPSHOT version of DSL

**Comment by karansaraswat19** [comment]
trigger ut5

**Comment by karansaraswat19** [comment]
trigger ut1

**Review by n00bitax — APPROVED** [review]

File: srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/cvng/utils/AzureUtils.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    if (type == VerificationType.LOG) {
+      return AZURE_LOGS_BASE_URL;
+    } else {
+      return AZURE_METRICS_BASE_URL;
+    }
+  }
+
+  public Map<String, String> collectionHeaders() {
+    Map<String, String> headers = new HashMap<>();
+    headers.put("content-type", "application/json");
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/cvng/utils/AzureUtils.java:33]: nit: is it suppose to start with lower case?

**Comment by n00bitax** [comment]
We are adding 0 test cases, we are prone to bugs when there will be a lot of changes. Please follow SumologicLogDataCollectionDSLTest and at least complete the equivalent of this class before adding anything else.
We should try to make our PRs one unit of "complete work".

**Comment by karansaraswat19** [comment]
trigger ut5

**Comment by deepak-harness** [comment]
trigger ut5

**Comment by karansaraswat19** [comment]
trigger uts

**Comment by karansaraswat19** [comment]
trigger ut8

**Comment by karansaraswat19** [comment]
trigger ut4

**Comment by karansaraswat19** [comment]
trigger ut8

File: srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/cvng/beans/azure/AzureLogsSampleDataRequest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  @Override
+  public Map<String, Object> fetchDslEnvVariables() {
+    AzureManualDetailsDTO azureManualDetailsDTO =
+        AzureUtils.validateConnectorConfigurationType(getConnectorConfigDTO().getCredential().getConfig());
+    AzureClientSecretKeyDTO azureClientSecretKeyDTO =
+        AzureUtils.validateConnectorAuthenticationType(azureManualDetailsDTO.getAuthDTO().getCredentials());
+    Map<String, Object> dslEnvVariables = super.fetchDslEnvVariables();
+    dslEnvVariables.put("clientId", azureManualDetailsDTO.getClientId());
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/cvng/beans/azure/AzureLogsSampleDataRequest.java:62]: nit: shouldn't few of these go to AbstractAzureDataRequest

**Comment by karansaraswat19** [comment]
trigger ut4

**Review by n00bitax — APPROVED** [review]

**Comment by n00bitax** [comment]
trigger uts_JDK11

**Review by n00bitax — APPROVED** [review]

**Pull request merged**